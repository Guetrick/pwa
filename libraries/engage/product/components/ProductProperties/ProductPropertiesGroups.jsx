import React from 'react';
import PropTypes from 'prop-types';
import { Accordion } from '@shopgate/pwa-ui-material';
import { ProductPropertiesRows } from './ProductPropertiesRows';
import { containerDense, accordion } from './ProductProperties.style';

/**
 * @param {Object} props The component props.
 * @returns {JSX}
 */
export function ProductPropertiesGroups({ properties, groups }) {
  return groups.map(group => (
    <div key={group} className={accordion}>
      <Accordion renderLabel={() => group} testId={`product-properties-group-${group}`}>
        <div className={containerDense}>
          <table>
            <tbody>
              <ProductPropertiesRows
                properties={properties.filter(property => property.displayGroup === group)}
              />
            </tbody>
          </table>
        </div>
      </Accordion>
    </div>
  ));
}

ProductPropertiesGroups.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.string).isRequired,
  properties: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};
