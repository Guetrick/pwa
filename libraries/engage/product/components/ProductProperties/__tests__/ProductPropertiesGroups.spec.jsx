import React from 'react';
import { shallow } from 'enzyme';
import { ProductPropertiesGroups } from '../ProductPropertiesGroups';

const properties = [
  {
    displayGroup: 'Group1',
  },
  {
    displayGroup: 'Group1',
  },
  {
    displayGroup: 'Group2',
  },
  {
    displayGroup: 'Group2',
  },
];

const groups = ['Group1', 'Group2'];

describe('<ProductPropertiesGroups />', () => {
  it('should render two groups of properties', () => {
    const wrapper = shallow(<ProductPropertiesGroups properties={properties} groups={groups} />);
    expect(wrapper.find('Accordion').length).toEqual(2);
    expect(wrapper.find('table').length).toEqual(2);
    expect(wrapper.find('ProductPropertiesRows').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});
