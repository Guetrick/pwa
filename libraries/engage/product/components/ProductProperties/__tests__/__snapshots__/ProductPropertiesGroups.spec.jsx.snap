// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesGroups /> should render two groups of properties 1`] = `
Array [
  <div
    className={
      Object {
        "data-css-fz7184": "",
      }
    }
    key="Group1"
  >
    <Accordion
      renderLabel={[Function]}
      testId="product-properties-group-Group1"
    >
      <div
        className={
          Object {
            "data-css-2c4onn": "",
          }
        }
      >
        <table>
          <tbody>
            <ProductPropertiesRows
              properties={
                Array [
                  Object {
                    "displayGroup": "Group1",
                  },
                  Object {
                    "displayGroup": "Group1",
                  },
                ]
              }
            />
          </tbody>
        </table>
      </div>
    </Accordion>
  </div>,
  <div
    className={
      Object {
        "data-css-fz7184": "",
      }
    }
    key="Group2"
  >
    <Accordion
      renderLabel={[Function]}
      testId="product-properties-group-Group2"
    >
      <div
        className={
          Object {
            "data-css-2c4onn": "",
          }
        }
      >
        <table>
          <tbody>
            <ProductPropertiesRows
              properties={
                Array [
                  Object {
                    "displayGroup": "Group2",
                  },
                  Object {
                    "displayGroup": "Group2",
                  },
                ]
              }
            />
          </tbody>
        </table>
      </div>
    </Accordion>
  </div>,
]
`;
